### PR TITLE
Features/widget test action button

### PR DESCRIPTION
**Pull request by NileshSingh9245**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-63

### Description

- 
1. **Test Case 1**: Verifies that both title and icon are displayed when both are provided to the `ActionButton`.

2. **Test Case 2**: Ensures that only the title is displayed when only a title is provided.

3. **Test Case 3**: Confirms that only the icon is displayed when only an icon is provided.

4. **Test Case 4**: Checks that the `onTap` callback is triggered when the `ActionButton` is tapped.

### Screenshots

WhatsApp Image 2024-12-17 at 00 45 30

### Tips

Any tips for testing it?

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'Staging' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin
- [x] All hard coded strings are present in constants file

### Primary reviewer checklist

- [ ] All items in the author checklist is checked or not applicable
- [ ] All files including .lock and assets are reviewed
- [ ] Reasonable UI flows/cases related to the change are manually tested
- [ ] No commented-out code or unnecessary debug statements.
- [ ] No spelling or grammatical errors in code comments or documentation.
- [ ] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [ ] The branch is up-to-date with the target branch (e.g., staging).
- [ ] The code adheres to established coding standards.

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

**Review by satyabratanayak — APPROVED** [review]

File: test/widgets/buttons/action_button_test.dart [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        ),
+      );
+
+      // Assert
+      expect(find.byIcon(testIcon), findsOneWidget);
+      expect(find.byType(Text), findsNothing);
+    });
+
+    testWidgets(
+        'When ActionButton is tapped '
```
Comment (quiteachaos) [inline test/widgets/buttons/action_button_test.dart]: add given statement here
Given ActionButton is rendered

Comment (NileshSingh9245) [reply test/widgets/buttons/action_button_test.dart]: testWidgets(
    'Given ActionButton is rendered '
    'When title and icon are provided '
    'Then it should display both title and icon',
    (WidgetTester tester) async {
  // Arrange
  await tester.pumpWidget(
    createWidgetUnderTest(
      title: testTitle,
      icon: testIcon,
    ),
  );

  // Assert
  expect(find.text(testTitle), findsOneWidget);
  expect(find.byIcon(testIcon), findsOneWidget);
});



Are you suggesting this  change?

Comment (quiteachaos) [reply test/widgets/buttons/action_button_test.dart]: yes like just add the given statement
given... when... then... is expected
do not change the rest of the code

Comment (NileshSingh9245) [reply test/widgets/buttons/action_button_test.dart]: Okay i will implement this change , Currently I don't have  access to my laptop.

**Pull request merged**